dali/python/nvidia/dali/plugin/tf.py

```diff
@@ -43,14 +43,17 @@
 :class:`~nvidia.dali.plugin.tf.DALIDataset`. This dataset adds support for input tf.data.Datasets.
 
 Each of the input datasets must be mapped to a :meth:`~nvidia.dali.fn.external_source` operator
-that will represent the input to the DALI pipeline. The input is represented by the ``name``
-parameter of :meth:`~nvidia.dali.fn.external_source` and needs to be provided for the corresponding
-input dataset via the ``input_names`` argument of DALIDatasetWithInputs.
-
-The input datasets are assumed to operate in sample mode - it means that every input dataset
-is treated as if providing individual samples.
-DALIDataset will query the inputs dataset ``batch_size``-times to build a batch that would
-be fed into the DALI Pipeline.
+that will represent the input to the DALI pipeline. In the pipeline the input is represented as
+the ``name`` parameter of :meth:`~nvidia.dali.fn.external_source`. Input datasets must be provided
+as a mapping from that ``name`` to the dataset object via the ``input_datasets`` dictionary argument
+of DALIDatasetWithInputs.
+
+The input datasets can operate in sample mode - it means that every input dataset
+is treated as if providing individual samples, or in batch mode - in that case the input dataset
+must return batches (Tensors with outermost dimension representing the sample index in batch).
```

Comment:
I think that the wording with "is treated as" is convoluted. Also - isn't the phrase "input dataset" used here to denote two different things?

Reply:
```suggestion
The input datasets can operate in per-sample mode or in batch mode.
In per-sample mode, the values produced by the source dataset are interpreted
as individual samples. The batch dimension is absent - for example, images would have HWC or CHW layout.
In batch mode, the tensors produced by the source dataset are interpreted as batches, with the
outermost dimension denoting the samples in the batch. A batch of images would have NHWC or NCHW layout.
```

Reply:
Done.